### PR TITLE
fix(validation): implement validation for  sign up

### PR DESCRIPTION
**Pull request by yvanddniyo**
**What does this PR do?**
This pull request implement  user validation for the sign up features and deny the extra fields for existing user fields.  User must provide valid email when a user provide invalid email the middleware deny it and return error. And also name of user can be lower than 5 characters as same as username can not go below 4  characters.

**Description of Task to be completed?**
Here are the endpoint working:

GET /api/v1/users
POST /api/v1/users/register

**How should this be manually tested?**

Clone the repository.
Checkout to the branch fix-userSignup-Validation#187431242.
Run npm install to install dependencies.
Run npm run dev to start the development server.
Open your postman or other your testing app and run http://localhost:3000/api/v1/users this allow you to retrieve existing user within in the database.
Use the endpoint of http://localhost:3000/api/v1/users/register to register a user using name, username, email and password.

**What are the relevant pivotal tracker stories.?**

#187431242

**Screenshot**

Screenshot (2)
Screenshot (3)
Screenshot (4)

### CODE REVIEW

File: src/controllers/userControllers.ts [hunk 1]
```diff
@@ -57,13 +58,21 @@ export const userLogin =  async(req:Request,res:Response) =>{
 
 export const createUserController = async (req: Request, res: Response) => {
   try {
-    const { name, email, username, password } = req.body;
+    const { value, error } = validateUser(req.body);
+    if (error) {
+      return res.status(400).json({ message: error.details[0].message });
+    }
```
Comment (teerenzo) [inline src/controllers/userControllers.ts]: make this helper function to be able to reuse it

**Review by teerenzo — CHANGES_REQUESTED** [review]

File: src/routes/userRoutes.ts [hunk 2]
```diff
@@ -4,12 +4,20 @@ import {
...
+ } from "../middleware/emailValidator";
 
 const userRoutes = Router();
 
 userRoutes.get("/", fetchAllUsers);
 userRoutes.post('/login',userLogin);
-userRoutes.post("/register", createUserController)
+userRoutes.post("/register", 
+ emailValidation, 
+ validateUserSignUp,
```
Comment (teerenzo) [inline src/routes/userRoutes.ts]: not like this @yvanddniyo this function will accept schema as parameter like this `validateSchema(signUpSchema)`

**Review by teerenzo — CHANGES_REQUESTED** [review]

File: src/validation/validate.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        .min(6)
+        .required()
+        .email(),
+    password: Joi.string()
+        .min(6)
+        .max(20)
+        .required()
+}).options({ allowUnknown: false });
+
+export default signUpSchema
```
Comment (teerenzo) [inline src/validation/validate.ts]: the folder should be something like schemas and also the file should be signUpSchema

Comment (teerenzo) [inline src/validation/validate.ts]: the folder should be something like schemas and also the file should be signUpSchema

**Review by teerenzo — CHANGES_REQUESTED** [review]

**Review by Jovz19200 — APPROVED** [review]
Well Done Yvan!

**Review by Heisjabo — APPROVED** [review]

File: src/schemas/signUpSchema.ts
Comment (soleil00) [inline src/schemas/signUpSchema.ts:1]: creating schema folder is unneccessary, u should define your schem inside validator middleware and use it inside,

**Review by soleil00 — APPROVED** [review]
except creating unneccessary schema your fix work as intended, u may consider to refactor ur code in near the future😌

**Review by teerenzo — APPROVED** [review]

**Pull request merged**